### ISSUE TITLE
Crucible backend needs a mechanism to be reconfigured before restarting after failed LM

### ISSUE THREAD

**Issue by gjcolombo** [oxidecomputer/propolis#230]
Hypothesized repro steps:

1. Launch a VM that connects to a set of Crucible downstairs with Crucible generation 1.
2. Start a migration target that will connect to the same downstairs with Crucible generation 2.
3. Inject a failure into migration immediately after the target activates (note that this can happen even after #155 is fixed if there is any way for the target to fail to start after Crucible activates).

Expected: The source will wait to be moved back to the 'Running' state. Before that happens, the control plane will set the source's Crucible generation to 3 and direct it to reactivate.

Current state: There are two problems to solve here:

- Nexus has no way to update the Crucible generation number of an ensured instance.
- The migration state machine automatically resumes the source after migration fails; it should wait until it's told to do so (and should allow configuration changes to be interposed between migration failure and source restart).

**Comment by leftwo** [comment]
> Nexus has no way to update the Crucible generation number of an ensured instance.

I'm working on the Omicron side to enable it to update a volume's generation number each time it is
pulled out of the database.  I think this might solve this issue, but what exactly to you want to 
happen to the "ensured instance"?  Is this "ensured_instance" in propolis, or Omicron?

**Comment by gjcolombo** [comment]
It's in Propolis. Specifically this code needs to change: https://github.com/oxidecomputer/propolis/blob/70e019a4e94e936f6ccd32656220a453d7e8c3b1/bin/propolis-server/src/lib/server.rs#L318-L344

Now that I look at it again, this issue is sort of a duplicate of #205, but that issue doesn't track the migration state machine problem I mentioned, so I'll keep this alive at least for that part of the problem.
